src/main/java/jeff/CommandHandler.java

```diff
@@ -0,0 +1,123 @@
+        int fromIndex = userInput.indexOf(" /from ") + 1;
+        int toIndex = userInput.indexOf(" /to ") + 1;
+        if (fromIndex == 0 || toIndex == 0 || fromIndex >= toIndex) {
+            throw new InvalidEventSyntaxException();
+        }
+        String description = userInput.substring(EVENT_DESCRIPTION_INDEX, fromIndex);
+        String from = userInput.substring(fromIndex + EVENT_FROM_INDEX_LENGTH, toIndex);
+        String to = userInput.substring(toIndex + EVENT_TO_INDEX_LENGTH);
+        if (description.isEmpty() || from.isEmpty() || to.isEmpty()) {
+            throw new InvalidEventSyntaxException();
+        }
+        Event event = new Event(description, from, to);
+        TaskList.add(event);
+        FileManager.appendTask(event);
+        Printer.printAddTask();
+    }
+
+    public static void handleMark(String userInput) throws InvalidMarkSyntaxException {
+        if (TaskList.isEmpty()) {
+            Printer.printUnableToMark();
+            return;
+        }
+        if (userInput.equals("mark")) {
+            throw new InvalidMarkSyntaxException();
+        }
+        try {
+            int currentIndex = Integer.parseInt(userInput.substring(MARK_INDEX)) - 1;
+            Task currentTask = TaskList.get(currentIndex);
+            currentTask.mark();
+            FileManager.updateMarkStatus(currentIndex, true);
+            Printer.printMarkTask(currentTask);
+        } catch (Exception e) {
+            throw new InvalidMarkSyntaxException();
+        }
+    }
+
+    public static void handleUnmark(String userInput) throws InvalidUnmarkSyntaxException {
+        if (TaskList.isEmpty()) {
+            Printer.printUnableToUnmark();
+            return;
+        }
+        if (userInput.equals("unmark")) {
+            throw new InvalidUnmarkSyntaxException();
+        }
+        try {
+            int currentIndex = Integer.parseInt(userInput.substring(UNMARK_INDEX)) - 1;
+            Task currentTask = TaskList.get(currentIndex);
+            currentTask.unmark();
+            FileManager.updateMarkStatus(currentIndex, false);
+            Printer.printUnmarkTask(currentTask);
+        } catch (Exception e) {
+            throw new InvalidUnmarkSyntaxException();
+        }
+    }
+
+    public static void handleDelete(String userInput) throws InvalidDeleteSyntaxException {
+        if (TaskList.isEmpty()) {
+            Printer.printUnableToDelete();
+            return;
+        }
+        if (userInput.equals("delete")) {
+            throw new InvalidDeleteSyntaxException();
+        }
+        try {
+            int currentIndex = Integer.parseInt(userInput.substring(UNMARK_INDEX)) - 1;
+            Task deletedTask = TaskList.remove(currentIndex);
+            FileManager.deleteTask(currentIndex);
+            Printer.printDeleteTask(deletedTask);
+        } catch (Exception e) {
+            throw new InvalidDeleteSyntaxException();
+        }
+    }
+
```

Comment:
the functions above have quite a few things in common, maybe to reduce redundancy, you can abstract out parts that are similar. 

https://nus-cs2113-ay2324s2.github.io/website/se-book-adapted/chapters/codeQuality.html#slap-hard